app/cdap/components/shared/ConfigurationGroup/index.tsx

```diff
@@ -85,6 +90,8 @@ const ConfigurationGroupView: React.FC<IConfigurationGroupProps> = ({
   }>({});
   const [filteredConfigurationGroups, setFilteredConfigurationGroups] = useState([]);
   const [orphanErrors, setOrphanErrors] = useState([]);
+  // a state to remember the properties values after each user interaction
+  const [currentValues, setCurrentValues] = useState(values);
```

Comment:
Seems like we're never calling `setCurrentValues`? Somehow I have a mixed feeling...

Reply:
Yea I chose to directly modify currentValues since `setCurrentValues` is async. Maybe there's a better way